lib/wasix/src/bin_factory/exec.rs

```diff
@@ -29,15 +29,25 @@ pub async fn spawn_exec(
     env: WasiEnv,
     runtime: &Arc<dyn Runtime + Send + Sync + 'static>,
 ) -> Result<TaskJoinHandle, SpawnError> {
+    // Spawn union the file system
+    spawn_union_fs(&env, &binary).await?;
+
     // Load the WASM
     let wasm = spawn_load_wasm(&env, &binary, name).await?;
 
+    spawn_exec_wasm(wasm, name, env, runtime).await
+}
+
+#[tracing::instrument(level = "trace", skip_all, fields(%name))]
+pub async fn spawn_exec_wasm(
+    wasm: &[u8],
+    name: &str,
+    env: WasiEnv,
+    runtime: &Arc<dyn Runtime + Send + Sync + 'static>,
+) -> Result<TaskJoinHandle, SpawnError> {
     // Load the module
     let module = spawn_load_module(&env, name, wasm, runtime).await?;
 
-    // Spawn union the file system
```

Comment:
not sure what this comment means, does it mean Spawn the union file system? I don't think we spawn it though we set it up

Reply:
You're right. This is a leftover comment from `spawn_exec`.